### PR TITLE
Meta: Add a note about commit hooks to the contribution guide

### PR DESCRIPTION
**Pull request by IdanHo**
These are pretty useful but are not documented anywhere, so this should fix that :^)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -88,3 +88,16 @@ In theory, the best person to speak with is whoever wrote most code adjacent to
...
+The repository contains a file called `.pre-commit-config.yaml` that defines several 'commit hooks' that can be run automatically just before and after creating a new commit. These hooks lint your commit message, and the changes it contains to ensure they will pass the automated CI for pull requests.
+To enable these hooks firstly follow the installation instructions available at https://pre-commit.com/#install and then enable one or both of the following hooks:
+ - pre-commit hook - Runs Meta/lint-ci.sh and Meta/lint-ports.py to ensure changes to the code will pass linting:
+   ```shell
+   pre-commit install
+   ```
+ - post-commit hook - Lints the commit message to ensure it will pass the commit linting:
+   ```shell
+   pre-commit install --hook-type commit-msg
+   ```
```
Comment (linusg) [inline CONTRIBUTING.md:103]: Just FYI, the proper language for "terminal commands" is `console`. :)

Comment (IdanHo) [reply CONTRIBUTING.md:103]: fixed :)

**Review by linusg — APPROVED** [review]
I didn't even know about `pre-commit install --hook-type commit-msg` :eyes:

**Comment by IdanHo** [comment]
> I didn't even know about `pre-commit install --hook-type commit-msg`

yep, added here: #6815, that just shows this section is useful :)

**Review by MaxWipfli — APPROVED** [review]
I wanted to do this for ages, thank you!

**Pull request merged**